src/lib/DepositApiClient.js

```diff
@@ -89,21 +89,33 @@ export class RDMDepositApiClient extends DepositApiClient {
   }
 
   async _createResponse(axiosRequest) {
+    let responseData = {};
+    let shouldThrowError = false;
+
     try {
       const response = await axiosRequest();
-      const data = this.recordSerializer.deserialize(response.data || {});
-      const errors = this.recordSerializer.deserializeErrors(
-        response.data.errors || []
-      );
-      return new DepositApiClientResponse(data, errors);
+      responseData = response.data;
     } catch (error) {
-      const data = this.recordSerializer.deserialize(error.response.data || {});
-      const errors = this.recordSerializer.deserializeErrors(
-        error.response.data.errors || []
-      );
-      const wrapped = new DepositApiClientResponse(data, errors);
-      throw new Error(wrapped);
+      shouldThrowError = true;
+      responseData = error.response.data;
     }
+
+    const data = this.recordSerializer.deserialize(responseData || {});
+
+    const noFormFieldErrors =
```

Comment:
Very minor: best practice is to go for "positive" vars, it is easier to understand while reading it.
```suggestion
    const formHasErrors = ...
```